container-engine-lib/lib/backend_impls/remote_context_backend/remote_context_kurtosis_backend.go

```diff
@@ -0,0 +1,208 @@
+package remote_context_backend
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/api_container"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/enclave"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/engine"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/exec_result"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/logs_collector"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/logs_database"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/networking_sidecar"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/kurtosis-tech/stacktrace"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"net"
+)
+
+// RemoteContextKurtosisBackend is a dual context holding a reference to a local backend running on Docker (k8s is
+// currently not supported to work with a remote Kurtosis backend) and a reference to a remote Kurtosis backend (which
+// currently is a Docker backend, we do not support anything else).
+// Basically, requests are dispatched among those 2 backends depending on which area of Kurtosis they touch.
+// This might change, but right now the engine runs locally, therefore engine interacting endpoint goes to the local
+// backend, and all the rest goes to the remote backend, as all other containers (including API container) should be
+// running in the remote backend
+type RemoteContextKurtosisBackend struct {
+	localKurtosisBackend  backend_interface.KurtosisBackend
+	remoteKurtosisBackend backend_interface.KurtosisBackend
```

Comment:
maybe we could be more explicit in the name that this is apic only

Reply:
Can do, but 1. I don't think that's even true. And 2. even if it was, on the long run it might become less and less true

1. It's not true because right now the CLI for example also uses this backend. 
2. We could change it to `localKurtosisBackendForEngineInteraction` and `remoteKurtosisBackendForEnclaveInteraction` because effectively it's what it is, but it is true only because we offload the entire enclave to the remote right now. Not even the APIC is local. However, I think we're soon going to have partial enclaves locally/remotely, and this will change. 

That's basically why I keep a somewhat generic name here. However you'll notice I explain that in the comment above. Let me know if that work, can also rename if you feel strongly about it

Reply:
I meant `forEngineInteraction` as well but I guess yours is more explicit

I think the current thing is alright.